src/state/contexts/TransactionState.tsx

```diff
@@ -0,0 +1,69 @@
+import {
+  Dispatch,
+  createContext,
+  useContext,
+  useEffect,
+  useReducer,
+} from 'react';
+
+import {
+  AntInteraction,
+  CONTRACT_TYPES,
+  ContractType,
+  REGISTRY_INTERACTION_TYPES,
+  RegistryInteraction,
+} from '../../types';
+import { TransactionAction } from '../reducers/TransactionReducer';
+
+export type TransactionState = {
+  transactionData: { [x: string]: any }; // data that will be used to perform the transaction.
```

Comment:
nit: be more specific about the transaction data type here, it should likely be `function: string` & `{ payload: ...the interaction payload types we'd expect`

Reply:
created type